### DISCUSSION TITLE
How to add external plugin to a kubernetes containerized instance

### DISCUSSION THREAD

**Discussion by reesericci** [coredns/coredns#4433] [Q&A] [answered]
I want to add the k8s_gateway plugin to my coredns instance but using kubernetes to manage my coredns. How do I add the external plugin to my instance?

Thanks,
Reese

P.S Enable discussions on github to allow for questions to be distinguished from issues.

**Comment by miekg** [comment, marked as answer]
[ Quoting <notifications@github.com> in "[coredns/coredns] How to add extern..." ]
>I want to add the k8s_gateway plugin to my coredns instance but using
>kubernetes to manage my coredns. How do I add the external plugin to my
>instance?

you have to compile it in https://coredns.io/2017/07/25/compile-time-enabling-or-disabling-plugins/

>P.S Enable discussions on github to allow for questions to be distinguished
>from issues.

good point. Done

**Comment by reesericci** [comment]
figured it out, I had to use make instead of go build

**Comment by reesericci** [comment]
migrate this issue to a discussion

**Comment by chrisohaver** [comment]
A more complete answer (but by no means detailed):
1. add external plugin to plugin.cfg
2. compile and build docker containers (see Makefile.release)
3. tag and push docker containers to _your_ docker repo (see Makefile.release)
4. update the Kubernetes coredns Deployment in your cluster to use the new tag in your repo

**Reply by reesericci** [reply]
I did this:
Fork coredns
Clone my new coredns
Edit plugin.cfg
Run make
docker build -t docker.pkg.github.com/username/coredns:tag .
docker push docker.pkg.github.com/usernamei/coredns:tag
And then use this image instead of the standard coredns image. It's gonna be a pain to rebase! (Github also requires auth to pull down images, so you've got to add registry credentials.)

**Reply by miekg** [reply]
technically you only need plugin.cfg to be different. 

There will only be changes in some generated files, so conflicts will be sparse, you can just pull master from upstream every so often
